assets/src/components/editing/elements/foreign/ForeignModal.tsx

```diff
@@ -0,0 +1,46 @@
+import { CommandContext } from 'components/editing/elements/commands/interfaces';
+import { Modal, ModalSize } from 'components/modal/Modal';
+import * as ContentModel from 'data/content/model/elements/types';
+import React, { useCallback } from 'react';
+
+interface Props {
+  onDone: (changes: Partial<ContentModel.Foreign>) => void;
+  onCancel: () => void;
+  model: ContentModel.Foreign;
+  commandContext: CommandContext;
+}
+export const ForeignModal = (props: Props) => {
+  const [lang, setLang] = React.useState(props.model.lang || '');
+  const onLangChange = useCallback((e) => {
+    setLang(e.target.value);
+  }, []);
+
+  return (
+    <Modal
+      title="Foreign Language Settings"
+      okLabel="Save"
+      cancelLabel="Cancel"
+      size={ModalSize.LARGE}
+      onCancel={() => props.onCancel()}
+      onOk={() => props.onDone({ lang })}
+    >
+      <div className="row">
+        <div className="col-12">
+          <p className="mb-4">
+            Set this text as belonging to a foreign language for screen readers to pronounce
+            properly. This has no visual effect for the learner. You can set a project-wide default
+            in your project settings.
+          </p>
+          <div className="popup__modalContent">
+            <form onSubmit={() => {}} id="popup__trigger_mode">
+              <div className="form-group">
+                <label>Target Language</label>
+                <input type="text" className="form-control" onChange={onLangChange} value={lang} />
```

Comment:
Can you easily replace this text input with a select whose options are the ISO 639-1 language codes?

Perhaps the rendered label could be "Language [code]", for example: "French [fr]", and of course the option value would simply be the code.

[**ISO 639-1 Language Codes**](https://www.w3docs.com/learn-html/html-language-codes.html)

Reply:
![image](https://user-images.githubusercontent.com/333265/193287689-71e99a9a-b89c-4b44-af82-1af664718360.png)

Added that in.

Reply:
Nice!